src/Elliptic/FirstOrderOperator.hpp

```diff
@@ -0,0 +1,75 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "NumericalAlgorithms/LinearOperators/Divergence.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace elliptic {
+
+/*!
+ * \ingroup ActionsGroup
+ * \ingroup DiscontinuousGalerkinGroup
+ * \brief Compute the bulk contribution to the operator represented by the
+ * `OperatorTag` applied to the `VarsTag`.
+ *
+ * This action computes \f$A(u)=-\partial_i F^i(u) + S(u)\f$, where \f$F^i\f$
+ * and \f$S\f$ are the fluxes and sources of the system of first-order PDEs,
+ * respectively. They are defined such that \f$A(u(x))=f(x)\f$ is the full
+ * system of equations, with \f$f(x)\f$ representing sources that are
+ * independent of the variables \f$u\f$. In a DG setting, boundary contributions
+ * can be added to \f$A(u)\f$ in a subsequent action to build the full DG
+ * operator action.
+ *
+ * We generally build the operator $A(u)$ to perform elliptic solver iterations.
+ * This action can be used to build operators for both the linear solver and the
+ * nonlinear solver iterations by providing the appropriate `OperatorTag` and
+ * `VarsTag`. For example, the `OperatorTag` for the linear solver would
+ * typically be `LinearSolver::Tags::OperatorAppliedTo`. It is the elliptic
+ * equivalent to the time derivative in evolution schemes. Instead of using it
+ * to evolve the variables, the iterative linear solver uses it to compute a
+ * better approximation to the equation \f$A(u)=f\f$ (as detailed above).
+ *
+ * With:
+ * - `operator_tag` = `db::add_tag_prefix<OperatorTag, VarsTag>`
+ * - `fluxes_tag` = `db::add_tag_prefix<::Tags::Flux, VarsTag,
+ * tmpl::size_t<Dim>, Frame::Inertial>`
+ * - `div_fluxes_tag` = `db::add_tag_prefix<Tags::div, fluxes_tag>`
+ * - `sources_tag` = `db::add_tag_prefix<::Tags::Source, VarsTag>`
+ *
+ * Uses:
+ * - DataBox:
+ *   - `div_fluxes_tag`
+ *   - `sources_tag`
+ *
+ * DataBox changes:
+ * - Modifies:
+ *   - `operator_tag`
+ */
+template <size_t Dim, template <typename> class OperatorTag, typename VarsTag>
+struct FirstOrderOperator {
```

Comment:
The documentation for this struct does not seem to be describing the correct thing.  It keeps talking about actions and describes an action interface, but this isn't an action.

Reply:
made some changes to the docs. Lacking a better name I refer to it as an _invokable_ now. I don't know if we have decided on a name for these classes that conform to `mutate_apply`.